src/DynamoCore/Core/Threading/UpdateGraphAsyncTask.cs

```diff
@@ -112,6 +112,10 @@ protected override AsyncTask.TaskMergeInstruction CanMergeWithCore(AsyncTask oth
             // Comparing to another UpdateGraphAsyncTask, verify 
             // that they are updating a similar set of nodes.
 
+            if ((graphSyncData.DeletedSubtrees != null && graphSyncData.DeletedSubtrees.Any()) ||
+                (theOtherTask.graphSyncData.DeletedSubtrees != null && theOtherTask.graphSyncData.DeletedSubtrees.Any()))
+                return TaskMergeInstruction.KeepBoth;
+
```

Comment:
Argh, we still have that `DeletedSubtrees == null` case, should have been always non-null. Next time.